### PR TITLE
🚧 Implement reply to selected text feature

### PR DESCRIPTION
**Pull request by dev-drprasad**
**Status**
- [x] WIP
- [ ] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- api
- hyperion (frontend)

ref #2969

### CODE REVIEW

File: src/components/message/index.js [hunk 1]
```diff
@@ -73,6 +73,7 @@ class Message extends Component<Props> {
       replyToMessage({
         threadId,
         messageId: message.id,
+        snippet: window.getSelection().toString() || null,
```
Comment (mxstbr) [inline src/components/message/index.js:81]: Ohh TIL about `window.getSelection`, that's very neat! My main question about this approach is: how can we make sure that whatever is quoted is actually in the message that's being quoted?

Comment (dev-drprasad) [reply src/components/message/index.js:81]: @mxstbr I commented about it in issue https://github.com/withspectrum/spectrum/issues/2969#issuecomment-393448330

File: src/components/chatInput/index.js [hunk 2]
```diff
@@ -40,8 +40,14 @@ import type { Dispatch } from 'redux';
 
 const QuotedMessage = connect()(
   getMessageById(props => {
+    console.log('props :', props);
     if (props.data && props.data.message) {
-      return <QuotedMessageComponent message={props.data.message} />;
+      return (
+        <QuotedMessageComponent
```
Comment (dev-drprasad) [inline src/components/chatInput/index.js:46]: @brianlovin @mxstbr i am modifying `QuotedMessage` to selected text. is it good idea ?? should we create seperate component like `QuotedSnippet` ??

File: src/components/chatInput/index.js [hunk 3]
```diff
@@ -89,7 +95,7 @@ type Props = {
   networkOnline: boolean,
   threadData?: Object,
   refetchThread?: Function,
-  quotedMessage: ?{ messageId: string, threadId: string },
+  quotedMessage: { messageId?: string, snippet?: string },
```
Comment (dev-drprasad) [inline src/components/chatInput/index.js:102]: @brianlovin @mxstbr  are we using any library for types ?

File: src/components/message/view.js [hunk 4]
```diff
@@ -160,7 +170,7 @@ const Action = (props: ActionProps) => {
             tipText={`Reply`}
             tipLocation={'top'}
             size={24}
-            onClick={replyToMessage}
+            onMouseDown={replyToMessage} // onClick deselects selected text
```
Comment (dev-drprasad) [inline src/components/message/view.js:173]: @mxstbr @brianlovin `onClick` deselects selected text, so we always get `null`. so, i had to use `onMouseDown`. what do you think ? is there any better way ?

**Pull request closed without merging**